pkg/services/shorturls/short_url_service.go

```diff
@@ -0,0 +1,64 @@
+package shorturls
+
+import (
+	"context"
+	"time"
+
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/registry"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+	"github.com/grafana/grafana/pkg/util"
+)
+
+func init() {
+	registry.RegisterService(&ShortURLService{})
+}
+
+type ShortURLService struct {
+	SQLStore *sqlstore.SqlStore `inject:""`
+}
+
+func (s *ShortURLService) Init() error {
+	return nil
+}
+
+func (s ShortURLService) GetFullURLByUID(ctx context.Context, user *models.SignedInUser, uid string) (string, error) {
```

Comment:
Since the method returns a path, and not a URL, it makes sense to rename it to GetPathByUID:
```suggestion
func (s ShortURLService) GetPathByUID(ctx context.Context, user *models.SignedInUser, uid string) (string, error) {
```